test/new-e2e/tests/remote-config/utils_test.go

```diff
@@ -8,37 +8,91 @@ package remoteconfig
 import (
 	_ "embed"
 	"fmt"
-	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
-	"github.com/stretchr/testify/assert"
+	"os"
+	"path"
+	"strings"
 	"testing"
 	"time"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+	"github.com/stretchr/testify/assert"
 )
 
 // assertLogsEventually will verify that a given `agentName` component's logs contain a pattern.
 // It will continually retry until the `expectedLogPattern` is found or the `maxRetries` is reached,
 // waiting `retryInterval` between each attempt.
 // If the `expectedLogPattern` is not found or an error occurs, the calling test will fail.
-func assertLogsEventually(t *testing.T, rh *components.RemoteHost, agentName string, expectedLogPattern string, waitFor time.Duration, tick time.Duration) {
+func assertAgentLogsEventually(t *testing.T, rh *components.RemoteHost, agentName string, expectedLogs []string, waitFor time.Duration, tick time.Duration) {
 	t.Helper()
+	foundLogs := make(map[string]bool, len(expectedLogs))
+	missingLogs := make([]string, len(expectedLogs))
+	// initially all logs are missing
+	copy(missingLogs, expectedLogs)
+	remoteLogsPath := fmt.Sprintf("/var/log/datadog/%s.log", agentName)
+	t.Logf("looking for logs in %s", remoteLogsPath)
 	assert.EventuallyWithTf(t, func(c *assert.CollectT) {
-		output, err := rh.Execute(fmt.Sprintf("cat /var/log/datadog/%s.log", agentName))
-		if assert.NoError(c, err) {
-			assert.Contains(c, output, expectedLogPattern)
+		// read agent logs
+		agentLogs, err := readRemoteFile(rh, remoteLogsPath)
+		if !assert.NoError(c, err) {
+			return
```

Comment:
I guess assert.NoError will raise a message so no need to create one here, correct?

Reply:
It would not: assert does not interrupt the go routine execution. I should use `require` that calls `FailNow`, but currently it panics and interrupts the Eventually. This will change as soon as testify merges [this PR](https://github.com/stretchr/testify/pull/1481)

Reply:
As a consequence we might have put a log message to understand the error. But it's unlikely to happen so I guess it's fine

Reply:
I think we log the error out of the `Eventually`. All failed assertions messages are saved in the `assert.CollectT` and logged on `Fail`

https://github.com/marshall-lee/testify/blob/master/assert/assertions.go#L312-L336